subnets/main.tf

```diff
@@ -14,3 +14,9 @@ resource "aws_subnet" "subnets" {
 
   tags = "${merge("${var.tags}",map("Name", "${var.project}-${var.visibility}-${var.role}-${element(data.aws_availability_zones.available.names, count.index)}", "Environment", "${var.environment}", "Project", "${var.project}", "Role", "${var.role}", "Visibility", "${var.visibility}"))}"
 }
+
+resource "aws_route_table_association" "subnet_association" {
+  count          = "${length(var.route_tables) >0 ? "${var.num_subnets}" : 0 }"
```

Comment:
on line 21 I used element instead of route_tables[count.index] because it handles the case you describe returning the element it has